File: 3.project-root-learning-machine/android-app/app/src/main/java/com/learningapp/data/model/Course.kt
package com.learningapp.data.model

data class Course(
    val id: String,
    val name: String,
    val description: String,
    val coverImage: String,
    val category: String,
    val difficulty: String,
    val status: String,
    val createdBy: String,
    val createdAt: Long,
    val updatedAt: Long
)

data class CourseDetail(
    val id: String,
    val name: String,
    val description: String,
    val coverImage: String,
    val category: String,
    val difficulty: String,
    val status: String,
    val units: List<LearningUnit>,
    val createdAt: Long,
    val updatedAt: Long
)

data class LearningUnit(
    val id: String,
    val courseId: String,
    val title: String,
    val order: Int,
    val contentType: String,
    val videoUrl: String?,
    val textContent: String?,
    val images: List<String>?,
    val duration: Int?,
    val knowledgePoints: List<String>?
)

data class PageResponse<T>(
    val content: List<T>,
    val totalElements: Long,
    val totalPages: Int,
    val size: Int,
    val number: Int
)

File: 3.project-root-learning-machine/web-admin/src/api/course.ts
import apiClient from './client'

export interface Course {
  id: string
  name: string
  description: string
  coverImage: string
  category: string
  difficulty: string
  status: string
  createdBy: string
  createdAt: number
  updatedAt: number
}

export interface CreateCourseRequest {
  name: string
  description: string
  coverImage: string
  category: string
  difficulty: string
}

export const courseApi = {
  getCourseList: (params?: { category?: string; status?: string; page?: number; size?: number }) => {
    return apiClient.get<{ content: Course[]; totalElements: number }>('/courses', { params })
  },
  
  getCourseDetail: (courseId: string) => {
    return apiClient.get(`/courses/${courseId}`)
  },
  
  createCourse: (data: CreateCourseRequest) => {
    return apiClient.post<Course>('/admin/courses', data)
  },
  
  updateCourse: (courseId: string, data: Partial<CreateCourseRequest>) => {
    return apiClient.put<Course>(`/admin/courses/${courseId}`, data)
  },
  
  publishCourse: (courseId: string) => {
    return apiClient.post(`/admin/courses/${courseId}/publish`)
  },
  
  deleteCourse: (courseId: string) => {
    return apiClient.delete(`/admin/courses/${courseId}`)
  }
}
